modin/numpy/arr.py

```diff
@@ -763,31 +744,33 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=None, *, where=True):
                     out._query_compiler = (
                         numpy.ones_like(out) * numpy.nan
                     )._query_compiler
-                return (
-                    fix_dtypes_and_determine_return(
+                if where is not False or out is not None:
+                    return fix_dtypes_and_determine_return(
                         result, 1, dtype, out, where is not False
                     )
-                    if where is not False or out is not None
-                    else array([numpy.nan], dtype=out_dtype)
-                )
+                else:
+                    return array([numpy.nan], dtype=out_dtype)
             return result.to_numpy()[0, 0] if where else numpy.nan
         if axis is None:
             result = self
             if isinstance(where, array):
                 result = where.where(self, numpy.nan)
+            # Since our current QueryCompiler does not have a mean that reduces 2D objects to
+            # a single value, we need to calculate the mean ourselves. First though, we need
+            # to figure out how many objects that we are taking the mean over (since any
+            # entries in our array that are `numpy.nan` must be ignored when taking the mean,
+            # and so cannot be included in the final division (of the sum over num total elements))
+            num_na_elements = (
+                result._query_compiler.isna().sum(axis=1).sum(axis=0).to_numpy()[0, 0]
+            )
+            num_total_elements = prod(self.shape) - num_na_elements
             result = (
                 numpy.array(
                     [result._query_compiler.sum(axis=1).sum(axis=0).to_numpy()[0, 0]],
                     dtype=out_dtype,
                 )
-                / (
-                    prod(self.shape)
-                    - result._query_compiler.isna()
-                    .sum(axis=1)
-                    .sum(axis=0)
-                    .to_numpy()[0, 0]
-                )
-            ).flatten()[0]
+                / num_total_elements
+            )[0]
```

Comment:
now I wonder what is the reason of making a single-value numpy array to only take its sole element straight away... cannot we do it like
```python
result = result._query_compiler.sum(axis=1).sum(axis=0).to_numpy()[0, 0] / num_total_elements
```
?

Reply:
I wanted to keep it a NumPy array so that we can get NumPy division rules - say we have 0 non-na elements in the list - rather than having an error (because we're dividing by 0), NumPy will handle the division correctly, and give us the appropriate result (i.e. NaN vs. INF depending on what our output dtype should be)